features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapSinkConsumer.java

```diff
@@ -0,0 +1,163 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.trapd;
+
+import static org.opennms.core.utils.InetAddressUtils.addr;
+
+import javax.annotation.PostConstruct;
+
+import org.opennms.core.ipc.sink.api.MessageConsumer;
+import org.opennms.core.ipc.sink.api.MessageConsumerManager;
+import org.opennms.core.ipc.sink.api.SinkModule;
+import org.opennms.core.utils.InetAddressUtils;
+import org.opennms.netmgt.config.TrapdConfig;
+import org.opennms.netmgt.config.api.EventConfDao;
+import org.opennms.netmgt.dao.api.DistPollerDao;
+import org.opennms.netmgt.dao.api.InterfaceToNodeCache;
+import org.opennms.netmgt.events.api.EventForwarder;
+import org.opennms.netmgt.model.events.EventBuilder;
+import org.opennms.netmgt.trapd.jmx.TrapdInstrumentation;
+import org.opennms.netmgt.xml.event.Event;
+import org.opennms.netmgt.xml.event.Events;
+import org.opennms.netmgt.xml.event.Log;
+import org.opennms.netmgt.xml.eventconf.Logmsg;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Qualifier;
+
+public class TrapSinkConsumer implements MessageConsumer<TrapInformationWrapper, TrapLogDTO> {
+
+	public static final TrapdInstrumentation trapdInstrumentation = new TrapdInstrumentation();
+
+	private static final Logger LOG = LoggerFactory.getLogger(TrapSinkConsumer.class);
+
+	/**
+	 * The name of the local host.
+	 */
+	private static final String LOCALHOST_ADDRESS = InetAddressUtils.getLocalHostName();
+
+	@Autowired
+	private MessageConsumerManager messageConsumerManager;
+
+	@Autowired
+	private EventConfDao eventConfDao;
+
+	@Autowired
+	@Qualifier("eventIpcManager")
+	private EventForwarder eventForwarder;
+
+	@Autowired
+	private InterfaceToNodeCache interfaceToNodeCache;
+
+	@Autowired
+	private TrapdConfig config;
+
+	@Autowired
+	private DistPollerDao distPollerDao;
+
+	private EventCreator eventCreator;
+
+	@PostConstruct
+	public void init() throws Exception {
+		messageConsumerManager.registerConsumer(this);
+		eventCreator = new EventCreator(interfaceToNodeCache, eventConfDao);
+	}
+
+	@Override
+	public SinkModule<TrapInformationWrapper, TrapLogDTO> getModule() {
+		return new TrapSinkModule(config, distPollerDao.whoami());
+	}
+
+	@Override
+	public void handleMessage(TrapLogDTO messageLog) {
+		final Log eventLog = toLog(messageLog);
```

Comment:
We should set the logging prefix for the thread here, so that log messages within this callback show up in trapd.log.

Here's how I do it in the Syslog implemenation: https://github.com/OpenNMS/opennms/blob/jira/HZN-966/features/events/syslog/src/main/java/org/opennms/netmgt/syslogd/SyslogSinkConsumer.java#L99